lib/Middleware/WOPIMiddleware.php

```diff
@@ -41,29 +42,31 @@
 use OCP\IConfig;
 use OCP\IRequest;
 use Psr\Log\LoggerInterface;
+use ReflectionClass;
+use ReflectionMethod;
 use Symfony\Component\HttpFoundation\IpUtils;
 
 class WOPIMiddleware extends Middleware {
-	/** @var IConfig */
-	private $config;
-	/** @var IRequest */
-	private $request;
-	/** @var WopiMapper */
-	private $wopiMapper;
-	/** @var LoggerInterface */
-	private $logger;
-
-	public function __construct(IConfig $config, IRequest $request, WopiMapper $wopiMapper, LoggerInterface $logger) {
-		$this->config = $config;
-		$this->request = $request;
-		$this->wopiMapper = $wopiMapper;
-		$this->logger = $logger;
+	public function __construct(
+		private IConfig $config,
+		private IRequest $request,
+		private WopiMapper $wopiMapper,
+		private LoggerInterface $logger
+	) {
 	}
 
 	public function beforeController($controller, $methodName) {
 		parent::beforeController($controller, $methodName);
 
-		if ($controller instanceof WopiController && !$this->isWOPIAllowed()) {
+		// Check controllers that are only supposed to be called by Collabora directly
+		$reflectionClass = new ReflectionClass($controller);
+		$hasClassAttribute = !empty($reflectionClass->getAttributes(RestrictToWopiServer::class));
+
+		$reflectionClass = new ReflectionMethod($controller, $methodName);
+		$hasMethodAttribute = !empty($reflectionClass->getAttributes(RestrictToWopiServer::class));
+
+		$isRestrictedController = $hasClassAttribute && $hasMethodAttribute;
```

Comment:
I think you want an `||` here, right?

Otherwise this would never apply, as the `WopiController` has the classAttribute but no method attributes and the get method of the `AssetsController` has method but no class attributes.

Reply:
```suggestion
		$isRestrictedController = $hasClassAttribute || $hasMethodAttribute;
```